lib/application/widgets/theme/rich_text_field.dart

```diff
@@ -0,0 +1,297 @@
+import 'dart:convert';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:flutter_quill/flutter_quill.dart' as quill;
+import 'package:layoutr/common_layout.dart';
+import 'package:memo/application/constants/dimensions.dart' as dimens;
+import 'package:memo/application/constants/images.dart' as images;
+import 'package:memo/application/constants/strings.dart' as strings;
+import 'package:memo/application/theme/theme_controller.dart';
+import 'package:memo/application/utils/bottom_sheet.dart';
+import 'package:memo/application/widgets/material/asset_icon_button.dart';
+import 'package:memo/application/widgets/theme/custom_button.dart';
+import 'package:memo/application/widgets/theme/themed_container.dart';
+import 'package:tuple/tuple.dart';
+
+/// A [TextField] variation that supports rich text content.
+///
+/// The field supports rich text content by opening a modal when tapped with an embedded `Quill Editor`.
+///
+/// The field also supports presenting rich text content, different from [TextField] which only supports simple
+/// text content. The field content height is limited by [dimens.richTextFieldConstraints].
+class RichTextField extends HookWidget {
+  const RichTextField({required this.title, required this.placeholder, this.controller});
+
+  /// The modal title positioned in the top-left corner of the modal editor.
+  final Widget title;
+
+  /// The placeholder used in the rich text editor when it has no content.
+  final String placeholder;
+
+  /// Controls the text content being edited.
+  final RichTextEditingController? controller;
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = useTheme();
+    final textTheme = Theme.of(context).textTheme;
+    final inputDecorationTheme = Theme.of(context).inputDecorationTheme;
+
+    final initialText = '${controller?.richTextContent ?? ''}\n';
+    final initialController = quill.QuillController(
+      document: quill.Document.fromDelta(quill.Delta()..insert(initialText)),
+      selection: const TextSelection.collapsed(offset: 0),
+    );
+    // Holds the entire field `quill` controller which is later replaced by `modalEditorController`, since it's not
+    // possible to forward the content from one controller to another.
+    final richFieldController = useState(initialController);
```

Comment:
I'm not sure about this, it seems completely bizarre, although I couldn't find any other way to forward `modalEditorController` content to `richFieldController`

Reply:
I figured out that having two controllers didn't make any sense at all. I just need to use the same `QuillController` in both instancies.